exercises/practice/binary-search-tree/src/BinarySearchTree.elm

```diff
@@ -0,0 +1,16 @@
+module BinarySearchTree exposing (BinaryTree(..), makeTree, sort)
+
+
+type BinaryTree
+    = Nil
```

Comment:
Recommend renaming `Nil` --> `Empty` to avoid confusion with Elm not supporting the concept of nil.